exporter/opentelemetry-exporter-otlp/tests/test_otlp_trace_exporter.py

```diff
@@ -157,6 +157,19 @@ def setUp(self):
     def tearDown(self):
         self.server.stop(None)
 
+    def test_gzip_compression(self):
+        mock_get_compression = mock.Mock()
+        patch = mock.patch(
+            "opentelemetry.exporter.otlp.trace_exporter",
+            side_effect=mock_get_compression,
+        )
+        compression = Compression.Gzip
+        with patch:
+            exporter = OTLPSpanExporter(compression="gzip")
+        
+        self.assertEqual(exporter.compression, compression)
```

Comment:
Would `exporter.compression` work? I thought it wasn't a property anymore.

Reply:
Yea sorry, I thought with Mock you could mock variables as well. Do you know how I can get access to the compression variable then?

Reply:
```
with mock.patch('grpc.insecure_channel`) as channel_mock:
   exporter = OTLPSpanExporter(compression="gzip")
   channel_mock.assert_called_with(""localhost:55680", Compression.Gzip)
```

Something like that.

Reply:
Could you take a look at the new test I added? For some reason, `insecure_channel()` is not being called. I tried to force call `insecure_channel()` by putting it outside the if statement for checking the credentials on line 160 in `exporter.py` but its still not getting called in the test.

Reply:
Are you sure `insecure_channel()` is not being called or is it just your `assert` that is failing? I would put some print statements or something in the exporter code to make sure that a channel is being created. If so, it could just be that `assert_called_with` is not the correct assert or that the mock patch isn't working as we expect. I would look into the docs/behaviour for both of those things.

If that's too much, you can always find some other way to test the behaviour. What we really only care about is: given whatever `compression` I provide to the constructor, the expected `compression_algorithm` is used. You should also test all possibilities of this (different `compression` values`, combinations of explicit vs environment variable, etc